### PR TITLE
POLIO-630 Fix timeout on calendar map merged shape- precalculate shape at update

### PR DESCRIPTION
**Pull request by olethanh**
Precalculate the geojson for each campaign at update time instead of when calling the API

## Self proof reading checklist

- [x] Did I use eslint and black formatters
- [x] Is my code clear enough and well documented
- [x] Are my typescript files well typed
- [NA] Did I add translations
- [x] My migrations file are included
- [?] Are there enough tests

## Changes

- Added `geojson` GeoJson field  on Campaign with it's shape (well a list of)
- Also rigged the admin so it's recalculate when updating the model from there
- and added a Admin action to recalculate shapes
- Added a migration

## How to test

With polio setup
Try to change the scope a campaign, multiple time
See if it is still displayed in the Calendar
Also in Anonymous/embedded mode

### CODE REVIEW

File: plugins/polio/api.py [hunk 1]
```diff
@@ -471,6 +471,7 @@ def return_cached_response_if_valid(cache_key, update_dates):
         url_path="v2/merged_shapes.geojson",
     )
     def shapes_v2(self, request):
+        "Deprecated, should return the same format as shapes v3, kept for comarison"
```
Comment (quang-le) [inline plugins/polio/api.py]: typo 😁

File: plugins/polio/serializers.py [hunk 2]
```diff
@@ -685,7 +689,10 @@ def update(self, instance: Campaign, validated_data):
 
     class Meta:
         model = Campaign
-        fields = "__all__"
+        # TODO in the future specicy the fields we actually want
```
Comment (quang-le) [inline plugins/polio/serializers.py]: typo lol

**Review by quang-le — COMMENTED** [review]
Seems roughly 2x faster on my machine 😀

**Comment by olethanh** [comment]
fixed the typois

**Comment by beygorghor** [comment]
@olethanh you have a migration conflict here

**Comment by olethanh** [comment]
I dont' understand how that happened since the other file was not even in my branch, seems like a strange github cache problem. Anyway fixed it my merging main and merging migrations

**Review by mathvdh — APPROVED** [review]
Il n'y a rien qui me choque mais je ne connais pas bien geojson ou les campaigns .. ?

**Review by niconoe — APPROVED** [review]
Same for me, not easy to understand all the global implications but the code looks fine IMHO.

**Pull request merged**